docs/specifications/tests/Zone-TP/zone11.md

```diff
@@ -0,0 +1,153 @@
+allowing domain name owners to specify which hosts are allowed to send mail
+claiming to be from that domain. It is implemented by means of TXT records in a
+structured format.
+
+This test looks up SPF records in the domain to be tested. It ensures that
+there is at most one such record and, if present, ensures it is syntactically
+valid and that its evaluation does not require more than 10 DNS lookups.
+
+## Scope
+
+It is assumed that *Child Zone* has been tested and reported by [Basic04]. This
+test case will just ignore non-responsive name servers or name servers not
+giving a correct DNS response for an authoritative name server.
+
+## Inputs
+
+* "Child Zone" - The domain name to be tested.
+
+## Summary
+
+Message Tag outputted           | Level   | Arguments            | Description of when message tag is outputted
+:-------------------------------|:--------|:---------------------|:--------------------------------------------
+SPF1_FOUND_AND_VALID            | INFO    |                      | The domain publishes a valid SPF version 1 policy.
+SPF1_LOOKUP_LIMIT_EXCEEDED      | ERROR   |                      | Evaluating the published SPF version 1 policy would require more than 10 DNS lookups.
+SPF1_MORE_THAN_ONE_SAME_MODIFIER | ERROR  | modifier             | The published SPF version 1 policy contains the same modifier (redirect or exp) more than once.
+SPF1_MULTIPLE_RECORDS           | ERROR   |                      | The domain publishes more than one SPF version 1 policy.
+SPF1_REDIRECT_AND_ALL           | WARNING |                      | The published SPF version 1 policy contains both the "all" keyword and a redirect modifier.
+SPF1_REDIRECT_NOT_AT_END        | WARNING |                      | The published SPF version 1 policy contains a redirect modifier, but it does not appear at the end of the string.
+SPF1_SYNTAX_ERROR               | ERROR   |                      | The published SPF version 1 policy has a syntax error.
+SPF_NOT_CONFIGURED              | INFO    |                      | No SPF policy was found.
+SPF_UNEXPECTED_ERROR            | ERROR   | rcode                | The server responded with an unexpected RCODE when looking up SPF records.
+SPF_WRONG_VERSION               | WARNING | version              | The domain publishes an SPF policy, but its version number is invalid.
+
+The value in the Level column is the default severity level of the message. The
+severity level can be changed in the [Zonemaster-Engine profile]. Also see the
+[Severity Level Definitions] document.
+
+The argument names in the Arguments column lists the arguments used in the
+message. The argument names are defined in the [argument list].
+
+## Test procedure
+
+1. Create a TXT query for the apex of the *Child Zone* and send it over UDP to
+   the name server.
+
+2. If the name server responds with an RCODE different from NOERROR, then
+   output *[SPF_UNEXPECTED_ERROR]* and terminate the test.
```

Comment:
Okay, so this makes a hypothetical `Z11_SPF_UNEXPECTED_ERROR` (and variants of these in the DMARC test) redundant.

Reply:
No, not necessarily. In [Nameserver02] the name server is tested how it handles EDNS queries. First an SOA query without EDNS is sent, and if the response is normal, a query with EDNS is sent and the behavior if this is anlyzed. Here a SOA query could be sent and you would expect the same RCODE if the owner name is the same.

[Nameserver02]: https://github.com/zonemaster/zonemaster/blob/develop/docs/specifications/tests/Nameserver-TP/nameserver02.md

Reply:
You are right: suppose all nameservers of a zone return SERVFAIL when we ask them for TXT records. Then the test would wrongfully emit a “no SPF configured” message. That would be misleading, because in fact we were unable to check for an SPF policy. That’s an issue I still need to take care of.

Reply:
I would say that it is unlikely that a server would that the server would correctly return an SOA record but not a TXT record.